awscli/customizations/s3/subcommands.py

```diff
@@ -533,15 +533,19 @@ def _display_page(self, response_data, use_basename=True,
                 filename_components = content['Key'].split('/')
                 filename = filename_components[-1]
             else:
-                filename = content['Key']
-            if strip_prefix:
-                if filename.startswith(strip_prefix):
-                    filename = filename[len(strip_prefix) + 1:]
+                filename = self._get_relative_key(key, content['Key'])
```

Comment:
I wonder if we can simplify this method a little more? Specifically, I think we have generalized the `_the_relative_key()` method enough that we may be able to just remove the `use_basename` completely and just make the `key` required.